### PR TITLE
Another attempt at GitHub Actions

### PR DESCRIPTION
**Pull request by d7415**
Some inspiration from #1068, plus extra locales, #1066 and a few other tweaks managed to get an almost-pass ([everything except pytz-latest](https://github.com/pimutils/khal/runs/3154445486?check_suite_focus=true#step:6:165)) before opening it as a PR. Let's see if it lasts...

Closes #1068
Closes #1013 
Closes #1066

### CODE REVIEW

**Comment by d7415** [comment]
Different failures each time. When they do fail they seem to be the same tests that have had issues previously (e.g. #860).

**Comment by d7415** [comment]
All of the failures switch between the 5 tests that use 
```py
vdirs[cal1].upload(DumbItem(
```

Those are:

- test_multi_uid_vdir
- test_event_different_timezones
- test_birthdays
- test_birthdays_29feb
- test_birthdays_no_year

**Comment by d7415** [comment]
It's a race condition.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -33,4 +39,5 @@ jobs:
           python -m pip install --upgrade coverage[toml] virtualenv tox tox-gh-actions          
 
       - name: "Run tox targets for ${{ matrix.python-version }}"
+        shell: 'script -q -e -c "bash --noprofile --norc -eo pipefail {0}"'
```
Comment (WhyNotHugo) [inline .github/workflows/ci.yml:42]: Do you mind leaving a comment as to what this does / why it's necessary?

I can imagine us looking at it in a year and having no idea :sweat_smile:

File: tests/khalendar_test.py [hunk 2]
```diff
@@ -339,9 +339,10 @@ def test_invalid_timezones(self, coll_vdirs):
         assert events[0].format('{start} {end} {title}', dt.date.today()) == \
             '02.12. 08:00 02.12. 09:30 Some event\x1b[0m'
 
-    def test_multi_uid_vdir(self, coll_vdirs, caplog, fix_caplog):
+    def test_multi_uid_vdir(self, coll_vdirs, caplog, fix_caplog, sleep_time):
         coll, vdirs = coll_vdirs
         caplog.set_level(logging.WARNING)
+        sleep(sleep_time)
```
Comment (WhyNotHugo) [inline tests/khalendar_test.py:345]: Are these required due to poor clock resolution on the CI environment or something?

**Review by WhyNotHugo — COMMENTED** [review]
Neat! Thanks for getting this working.

Just some questions on in, but LG.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (d7415) [reply .github/workflows/ci.yml:42]: Good thinking!

File: tests/khalendar_test.py (hunk 2 above)
Comment (d7415) [reply tests/khalendar_test.py:345]: I'm not sure about the details, but when I traced the failures I found that `needs_update()` was false, and on further investigation the ctag in the DB and the filesystem were returning the [same timestamp](https://github.com/pimutils/khal/runs/3156981790?check_suite_focus=true#step:6:183) (to what *looks* like high resolution).

I assume that the DB ctag is being created/reset at the start of the test, and if the upload step is faster than the clock resolution the test fails. The sleep used elsewhere in `khalendar_test.py` (~0.011s) seems to be enough to avoid this.

I guess I should comment these too...

**Comment by d7415** [comment]
I've also change it so that the mypy and style tests run as separate jobs, so in parallel and easier to see vs the normal py39 tests.

I reduced some duplication in tox.ini in https://github.com/pimutils/khal/commit/53b3fd5bbfbaa7d1a3ea68240f3502589daae55d but I think that sacrifices readability.

File: tests/khalendar_test.py (hunk 2 above)
Comment (WhyNotHugo) [reply tests/khalendar_test.py]: Note that they all end in `.000000000`, so it seems they're not high resolution. I wonder what filesystem is being used under the hood that has this issue. AFAIK, we've never seen anything this low resolution.

Some background:

`needs_update` will only return `True` if the file has changed since we last read it. If the clock is **too** low precision, then when we edit the file a second time, it's mtime hasn't changed (since the clock hasn't moved forward due to its lack of precision).

In simpler terms: if the clock only has milisecond precision, then we must wait at least 1.1 miliseconds between edits for the timestamp to change. `sleep_time` determines how much time we need to sleep (see its docs for details).

**Review by WhyNotHugo — APPROVED** [review]
LGMT.
